storage/namespace.go

```diff
@@ -547,12 +594,14 @@ func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
 		if shard == nil {
 			continue
 		}
-		state := shard.FlushState(blockStart)
-		if state.Status == fileOpNotStarted {
-			return true
-		}
-		if state.Status == fileOpFailed && state.NumFailures < maxRetries {
-			return true
+		for _, blockStart := range blockStarts {
```

Comment:
Same here, seems if there is a single shard that's failed or not flushed for any start time within the range, we'll reflush all shards for all start times within that range, which is also a bit strange.